### ISSUE TITLE
osmcontroller.addmarker function not working in ios

### ISSUE THREAD

**Issue by jesussmile** [liodali/osm_flutter#369]
In android this function works well but in ios the markers don't show. in general the `.addMarker` function is not working in ios

```ruby
 try {
    Future.microtask(() async {
        await widget.mapController!.removeLastRoad();
        GeoPoint pointA = appData.initPosition;

        GeoPoint pointB = appData.dropOffLatLng!;

        await widget.mapController!.addMarker(
            pointB,
            markerIcon: MarkerIcon(
                iconWidget: SvgPicture.asset(
                    "assets/pick_up.svg",
                    height: 80,
                ),
            ),
        );

        await widget.mapController!.addMarker(pointA,
            markerIcon: MarkerIcon(
                iconWidget: SvgPicture.asset(
                    "assets/pick_up.svg",
                    height: 80,
                ),
            ));

        RoadInfo roadInformation =
            await widget.mapController!.drawRoad(
                pointA,
                pointB,
                roadOption: RoadOption(
                    roadWidth: 10,
                    roadColor: Colors.purple,
                    showMarkerOfPOI: true,
                ),
            );
        my_loc_controller.forward();
        my_loc_controller.repeat();
        appData.showMapMainScreen();

        appData.showMapPriceDistance(
            roadInformation.distance!);
    }, );
```

IOS
<a href="https://ibb.co/sjKRN9Q"><img src="https://i.ibb.co/0qC26yZ/IMG-0002.jpg" alt="IMG-0002" border="0"></a>

ANDROID

<a href="https://ibb.co/XpSWFgS"><img src="https://i.ibb.co/d6GfkqG/20230129-110558.jpg" alt="20230129-110558" border="0"></a><br /><a target='_blank' href='https://imgbb.com/'>photo sharing services reviews</a><br />

Even the `defaultMarker` and `directionArrowMarker` in `OSMFlutter` widget wont show in IOS

```ruby
  markerOption: MarkerOption(
        defaultMarker: MarkerIcon(
          iconWidget: Image.asset(
            "images/images/gps.png",
            width: 54.w,
            height: 32.h,
          ),
        ),
      ),

      userLocationMarker: UserLocationMaker(
        directionArrowMarker: MarkerIcon(
          iconWidget: Image.asset(
            "images/images/gps.png",
            width: 54.w,
            height: 32.h,
          ),
        ),
```

**Comment by liodali** [comment]
i will check that

**Comment by jesussmile** [comment]
yes `flutter_osm_plugin: ^0.42.0`

**Comment by liodali** [comment]
try to add keep `keepInitialGeoPoints` : false

**Comment by jesussmile** [comment]
oh! works

**Comment by jesussmile** [comment]
I noticed one more thing `directionArrowMarker` and ` defaultMarker` is only visible when i set `trackMyPosition: true,` in ios but in android it works without it, The issue is the map will always center if I do that in ios where as in android it is fine

```ruby
OSMFlutter(
      //isPicker: true,
      key: _mapKey,
      controller: osmController!,
      initZoom: 18,
      minZoomLevel: 8,
      maxZoomLevel: 18,
      stepZoom: 1.0,
      trackMyPosition: true,
      roadConfiguration: RoadConfiguration(
        startIcon: MarkerIcon(
          iconWidget: Icon(
            Icons.location_on,
            color: Colors.red,
          ),
          //Lottie.asset("my_location.json"),
        ),
        endIcon: MarkerIcon(
          iconWidget: Icon(Icons.pin_drop),
          //iconWidget: Lottie.asset("my_location.json"),
        ),
        roadColor: Colors.yellowAccent,
      ),
      markerOption: MarkerOption(
        defaultMarker: MarkerIcon(
          iconWidget: Image.asset(
            "images/images/gps.png",
            width: 54.w,
            height: 32.h,
          ),
        ),
      ),

      userLocationMarker: UserLocationMaker(
        directionArrowMarker: MarkerIcon(
          iconWidget: Image.asset(
            "images/images/gps.png",
            width: 54.w,
            height: 32.h,
          ),
        ),
        personMarker: MarkerIcon(
          iconWidget: Image.asset(
            "images/images/gps.png",
            width: 54.w,
            height: 32.h,
          ),
        ),
      ),
    );
```

**Comment by liodali** [comment]
i will fix that since in ios we made track and marker manually not like in android

**Comment by liodali** [comment]
but directionArrow will be visible when controller.enableTracking called or also direction not visible

**Comment by liodali** [comment]
did you face the issue on simulator or real device ?

**Comment by jesussmile** [comment]
Real device

**Comment by liodali** [comment]
ios version is +16?

**Comment by jesussmile** [comment]
Ios 15 +

**Comment by liodali** [comment]
thnx because also in the simulator i hade the issue of arrow/person marker but in my iphone is running on 14.6 it working fine
so i need to find the issue why the icon not visible

**Comment by jesussmile** [comment]
I have a device on lower ios version, will check on it and get back to you.

**Comment by liodali** [comment]
technically i found the issue i will fix it 
the fix will be in `0.50.0` i will do my best to published soon

**Comment by liodali** [comment]
I will remove that parameter and provide another api to clear markers
